### PR TITLE
Playwright: Create custom fixtures to help interact with elements

### PR DESCRIPTION
**Pull request by ardelato**
## Description

By using Playwright custom fixtures[^1] we will be creating `iFixit/ifixit`'s extension-like classes to help with interacting with elements and making assertions.

## CR Notes

We will be adding more methods and fixtures in the future to fully address #1217, but for now this is a good start.

## QA

- [ ] CI Passes

Connects: https://github.com/ifixit/react-commerce/issues/1217

[^1]: https://playwright.dev/docs/test-fixtures#creating-a-fixture

qa_req 0

### CODE REVIEW

File: frontend/tests/playwright/product-list/collections_scroll.spec.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
-import { test, expect } from '@playwright/test';
+import { test, expect } from '../test-fixtures';
 // Assert that when a user clicks the pagination buttons,
 // the page scrolls back up.
 
 test.describe('collections scroll', () => {
-   test.beforeEach(async ({ page }) => {
+   test.beforeEach(async ({ page, productPage }) => {
```
Comment (batbattur) [inline frontend/tests/playwright/product-list/collections_scroll.spec.ts:6]: I think `productPage` in most of the collection tests is unused.

**Review by batbattur — DISMISSED** [review]
CR 🚛 awesome stuff @ardelato! 

deploy_block 🚚 on removing unused `productPage` variables in collection page tests.

File: frontend/tests/playwright/product-list/collections_scroll.spec.ts (hunk 1 above)
Comment (ardelato) [reply frontend/tests/playwright/product-list/collections_scroll.spec.ts]: Addressed in [`2245ca5` (#1258)](https://github.com/iFixit/react-commerce/pull/1258/commits/2245ca5bbdae207f0e9ea7bb5da0dd4abf376c2f)

**Comment by ardelato** [comment]
un_deploy_block ⚡

**Review by batbattur — APPROVED** [review]
CR 🚛 💨

**Pull request merged**